Doc/library/sqlite3.rst

```diff
@@ -2270,13 +2344,72 @@ can be found in the `SQLite URI documentation`_.
 Explanation
 -----------
 
+.. _sqlite3-transaction-control:
 .. _sqlite3-controlling-transactions:
 
 Transaction control
 ^^^^^^^^^^^^^^^^^^^
 
-The :mod:`!sqlite3` module does not adhere to the transaction handling recommended
-by :pep:`249`.
+:mod:`!sqlite3` offers multiple methods of controlling whether,
+when and how database transactions are opened and closed.
+:ref:`sqlite3-transaction-control-autocommit` is recommended,
+while :ref:`sqlite3-transaction-control-isolation-level`
+retains the pre-Python 3.12 behaviour.
+
+.. _sqlite3-transaction-control-autocommit:
+
+Transaction control via the ``autocommit`` attribute
+""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+The recommended way of controlling transaction behaviour is through
+the :attr:`Connection.autocommit` attribute,
+which should preferrably be set using the *autocommit* parameter
+of :func:`connect`.
+
+It is suggested to set *autocommit* to ``False``,
+which implies :pep:`249`-compliant transaction control.
+This means:
+
+* :mod:`!sqlite3` ensures that a transaction is always open,
+  so :meth:`Connection.commit` and :meth:`Connection.rollback`
+  will implicitly open a new transaction immediately after closing
+  the pending one.
+  :mod:`!sqlite3` uses ``BEGIN DEFERRED`` statements when opening transactions.
+* Transactions should be committed explicitly using :meth:`!commit`.
+* Transactions should be rolled back explicitly using :meth:`!rollback`.
+* An implicit rollback is performed if the database is
+  :meth:`~Connection.close`-ed with pending changes.
+
+Set *autocommit* to ``True`` to enable SQLite's `autocommit mode`_.
+In this mode, :meth:`Connection.commit` and :meth:`Connection.rollback`
+have no effect.
+Note that SQLite's autocommit mode is distinct from
+the :pep:`249`-compliant :attr:`Connection.autocommit` attribute;
+use :attr:`Connection.in_transaction` to query
+the low-level SQLite autocommit mode.
```

Comment:
```suggestion
Set *autocommit* to ``True`` to enable SQLite's `autocommit mode`_.
This means:

* :mod:`!sqlite3` does not implicitly open any transactions at all.
* The user can perform their own transaction handling
  with `BEGIN`, `COMMIT`, and `ROLLBACK` statements.
* SQLite's `autocommit mode`_ can be queried using the
  :attr:`~Connection.in_transaction` attribute.
* :meth:`Connection.commit` and :meth:`Connection.rollback`
  have no effect.
```

Reply:
- We should not encourage users to do their own transaction handling. `sqlite3` is _not_ an SQLite shim (like apsw); `sqlite3` is a _PEP 249 compliant_ interface. In PEP 249 there are only _two recommended modes of operation_: `autocommit is True` and `autocommit is False`.
- `Connection.in_transaction` is valid no matter the transaction mode of sqlite3. It does not make sense to list it as a specific thing for `autocommit is True`.
- The text hammered out by Alex and CAM is fine as it is.

Reply:
> We should not encourage users to do their own transaction handling.

That is a valid point, users might interpret this possibility as an advice.

> Connection.in_transaction is valid no matter the transaction mode of sqlite3.

I agree, the only reason I mention it here is because it was already in the original text, in the same paragraph as `Set *autocommit* to ``True```:

> use :attr:`Connection.in_transaction` to query
> the low-level SQLite autocommit mode.